.github/workflows/deploy.yml

```diff
@@ -25,27 +25,24 @@ jobs:
         with:
           terraform_version: 0.13.3
 
-      - name: Terraform init
-        run: |
-          terraform init -backend-config=workspace_variables/${{ env.DEPLOY_ENV }}_backend.tfvars
+      - name: Download app configuration file
         working-directory: terraform
+        run: |
+          echo $APP_SECRETS | base64 -d >> app_secrets.yml
         env:
-          ARM_ACCESS_KEY: ${{ secrets.TERRAFORM_STATE_ACCESS_KEY }}
-      
-      - name: Terraform plan & apply
+          APP_SECRETS: ${{ secrets[format('APP_SECRETS_{0}', env.DEPLOY_ENV)] }}
+
+      - name: Terraform init, plan & apply
         id: terraform_deploy
         run: |
+          terraform init -backend-config=workspace_variables/${{ env.DEPLOY_ENV }}_backend.tfvars
```

Comment:
You can use `${DEPLOY_ENV}` as the variable is available in the shell environment